packages/suite/src/views/settings/SettingsDevice/Homescreen.tsx

```diff
@@ -72,7 +73,11 @@ export const Homescreen = ({ isDeviceLocked }: HomescreenProps) => {
 
     const onUploadHomescreen = async (files: FileList | null) => {
         if (!files || !files.length) return;
-        const file = files[0];
+        let file = files[0];
+
+        // Do NOT touch the image if it's already valid
+        if (await validateImage({ file, deviceModelInternal }))
+            file = (await convertImage({ file, deviceModelInternal })) ?? file;
 
         const validationResult = await validateImage({ file, deviceModelInternal });
         setValidationError(validationResult);
```

Comment:
We should avoid repeated validation in case it passes the first time. You can use this or come up with something better:

```suggestion
        const validate = async () => await validateImage({ file, deviceModelInternal });
        let validationResult = await validate();

        // Do NOT touch the image if it's already valid
        if (validationResult) {
            file = (await convertImage({ file, deviceModelInternal })) ?? file;
            validationResult = await validate();
        }
        
        setValidationError(validationResult);
```

Reply:
Fixed in 3aea4cc